src/integrationTest/java/uk/gov/hmcts/reform/cpo/repository/CasePaymentOrdersAuditRepositoryTest.java

```diff
@@ -0,0 +1,261 @@
+import uk.gov.hmcts.reform.cpo.data.CasePaymentOrderEntity;
+import uk.gov.hmcts.reform.cpo.utils.CasePaymentOrderEntityGenerator;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class CasePaymentOrdersAuditRepositoryTest extends BaseTest {
+
+    @Autowired
+    CasePaymentOrdersAuditJpaRepository casePaymentOrdersAuditRepository;
+
+    @Autowired
+    CasePaymentOrdersJpaRepository casePaymentOrdersRepository;
+
+    @Autowired
+    private CasePaymentOrderEntityGenerator casePaymentOrderEntityGenerator;
+
+    @Autowired
+    private RevInfoRepository revInfoRepository;
+
+    @BeforeEach
+    public void setUp() {
+        casePaymentOrdersRepository.deleteAllInBatch();
+        casePaymentOrdersAuditRepository.deleteAllInBatch();
+    }
+
+    @Nested
+    @DisplayName("delete from case-payment-orders-audit table using UUIDs")
+    class DeleteCasePaymentOrdersByUUIDs {
+
+        /**
+         * Delete a single ID - Audit entry has one row for that ID.
+         */
+        @DisplayName("test deletion of an audit entity that has a single corresponding row from audit table")
+        @Test
+        public void testDeleteEntitySingleIdMatches() {
+            // Given
+            List<CasePaymentOrderEntity> persistedEntities =
+                    casePaymentOrderEntityGenerator.generateAndSaveEntities(1);
+            assertEquals(persistedEntities.size(), casePaymentOrdersAuditRepository.findAll().size());
+
+            // When
+            casePaymentOrdersAuditRepository.deleteByIdIn(List.of(persistedEntities.get(0).getId()));
+
+            // Then
+            assertTrue(casePaymentOrdersAuditRepository.findAll().isEmpty());
+            assertTrue(revInfoRepository.findAll().isEmpty());
+        }
+
+        /**
+         * Deleting multiple IDs - Audit entry has one row for each ID.
+         */
+        @DisplayName("test deletion of multiple audit entities that each have a single row, from audit table")
+        @Test
+        public void testDeleteEntitiesSingleIdMatches() {
+            // Given
+            List<CasePaymentOrderEntity> persistedEntities =
+                    casePaymentOrderEntityGenerator.generateAndSaveEntities(3);
+            assertEquals(persistedEntities.size(), casePaymentOrdersAuditRepository.findAll().size());
+
+            // When
+            casePaymentOrdersAuditRepository.deleteByIdIn(List.of(persistedEntities.get(0).getId(),
+                                                                    persistedEntities.get(1).getId()));
+
+            // Then
+            assertEquals(persistedEntities.size() - 2, casePaymentOrdersAuditRepository.findAll().size());
+            assertEquals(1, revInfoRepository.findAll().size());
+        }
+
+        /**
+         * Dleting a single ID - Audit entry has multiple rows for that ID.
```

Comment:
spotting a typo `Dleting` at least shows I read through the file.

Reply:
Fixed